recipes/redradist-icc/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class ICCConan(ConanFile):
+    name = 'redradist-icc'
+    homepage = 'https://github.com/redradist/Inter-Component-Communication'
+    license = 'MIT'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "I.C.C. - Inter Component Communication, This is a library created to simplify communication between " \
+                  "components inside of single application. It is thread safe and could be used for creating " \
+                  "components that works in different threads. "
+    topics = ("thread-safe", "active-object", "communication")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True, False],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "apple-clang": "9.4",
+            "clang": "3.3",
+            "gcc": "4.9.4"
+        }
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions['ICC_BUILD_SHARED'] = self.options.shared
+        cmake.configure()
+        self._cmake = cmake
+        return self._cmake
+
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+
+        os = self.settings.os
+        if os not in ("Windows", "Linux"):
+            msg = (
```

Comment:
Can we skip this variable?

Reply:
I am not sure why you done this, but it seems extra... can it be removed everywhere?

Reply:
Because it makes access to object shorter ... )

Reply:
it's 4 lines instead of one.... length might be relative (there's no action required just to be clear just my personal preference)

(please laugh at my gif)
![relative](https://media2.giphy.com/media/ZBKJ1VGLhubqPSZNUx/giphy.gif)